src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -286,6 +286,16 @@ protected virtual void Dispose(bool finalize)
             if (!finalize)
             {
                 this.ClearSolutionData();
+
+                // tell event listener to stop listening since this workspace is going away
+                var workspaceEventListenerProvider = this.Services.GetService<IWorkspaceEventListenerProvider>();
+                if (workspaceEventListenerProvider != null)
+                {
+                    foreach (var listener in workspaceEventListenerProvider.GetListeners())
+                    {
+                        listener.Stop(this);
```

Comment:
We call stop even if we haven't initialized in the first place?

Reply:
changed.